ports/atmel-samd/common-hal/pulseio/PulseIn.c

```diff
@@ -283,10 +283,11 @@ void common_hal_pulseio_pulsein_resume(pulseio_pulsein_obj_t* self,
     if (trigger_duration > 0) {
         gpio_set_pin_pull_mode(self->pin, GPIO_PULL_OFF);
         gpio_set_pin_direction(self->pin, GPIO_DIRECTION_OUT);
-
-        common_hal_mcu_disable_interrupts();
+        if(trigger_duration < 1000 ){  // only disable interrupts for < 1 ms
+            common_hal_mcu_disable_interrupts();
+        }
```

Comment:
Since tick_delay is interrupt agnostic we can delete this too. (Unless we want to guarantee precise timing.)

Reply:
Are you suggesting that it is not necessary to disable interrupts at all for pulseio? If so , fine we can get rid of this.  If we want "precise" trigger_durations when < 1000 us then we still ned it.

Reply:
Yeah, I'm debating that. Either way is fine for now.